src/main/java/fi/aalto/cs/apluscourses/ui/utils/TreePathCoder.java

```diff
@@ -0,0 +1,23 @@
+package fi.aalto.cs.apluscourses.ui.utils;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.swing.tree.TreePath;
+import org.jetbrains.annotations.Nullable;
+
+public abstract class TreePathCoder<T> {
```

Comment:
What's a tree path coder?

Reply:
I think I get it now, but would the more correct word be `Encoder` then?

Reply:
Yes. English difficult.